### PR TITLE
Add Android client

### PR DESCRIPTION
**Pull request by janvennemann**

### CODE REVIEW

File: android/android.iml [hunk 1]
```diff
@@ -7,5 +7,11 @@
     </content>
     <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="library" name="socket.io-client-1.0.0" level="project" />
+    <orderEntry type="library" name="titanium" level="project" />
+    <orderEntry type="library" name="kroll-common" level="project" />
+    <orderEntry type="library" name="kroll-apt" level="project" />
+    <orderEntry type="library" name="engine.io-client-1.0.0" level="project" />
+    <orderEntry type="library" name="json-20180130" level="project" />
```
Comment (hansemannn) [inline android/android.iml]: What does this do?

File: android/timodule.xml [hunk 2]
```diff
@@ -7,5 +7,8 @@
 	<iphone>
 	</iphone>
 	<android xmlns:android="http://schemas.android.com/apk/res/android">
+		<manifest>
+			<uses-permission android:name="android.permission.INTERNET" />
```
Comment (hansemannn) [inline android/timodule.xml]: Don't we add this permission [already](https://github.com/appcelerator/titanium_mobile/blob/master/android/cli/commands/_build.js#L3615) ?

File: android/manifest [hunk 3]
```diff
@@ -15,4 +15,4 @@ name: ti.socketio
 moduleid: ti.socketio
 guid: 65fbdc59-432f-40d8-8ea9-02799887db3c
 platform: android
-minsdk: 7.2.0.v20180423082013
+minsdk: 7.1.0.GA
```
Comment (hansemannn) [inline android/manifest]: We should probably allow this module to be used from 7.0.0+ already.

File: android/src/com/appc/titanium/socketio/SocketIOClientProxy.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+	public SocketIOClientProxy(Socket socket)
+	{
+		super();
+
+		this.socket = socket;
+		this.callbackListenerMap = new WeakHashMap<>();
+	}
+
+	@Kroll.getProperty
+	public boolean getConnected()
```
Comment (hansemannn) [inline android/src/com/appc/titanium/socketio/SocketIOClientProxy.java]: `@Kroll.method` as well, to access it via `getConnected()` ?

File: android/src/com/appc/titanium/socketio/SocketIOClientProxy.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
+/**
+ * This file was auto-generated by the Titanium Module SDK helper for Android
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2017 by Appcelerator, Inc. All Rights Reserved.
```
Comment (hansemannn) [inline android/src/com/appc/titanium/socketio/SocketIOClientProxy.java]: 2018 or present.

Comment (janvennemann) [reply android/src/com/appc/titanium/socketio/SocketIOClientProxy.java]: We should probably check the generator then. It should not have generated a file with 2017 as the copyright range.

File: android/src/com/appc/titanium/socketio/SocketIOClientProxy.java (hunk 4 above)
Comment (janvennemann) [reply android/src/com/appc/titanium/socketio/SocketIOClientProxy.java]: IMHO there is no benefit for an additional getter so i just stayed with the property access to `connected`.

**Comment by hansemannn** [comment]
We should change the [iOS manifest](https://github.com/appcelerator-modules/titanium-socketio/blob/master/ios/manifest#L18) to use 7.0.0 as well.

**Comment by janvennemann** [comment]
@hansemannn Done! See 2731b70ef0ab8edc14f2fb4a7c47a76b96a2954b

**Comment by hansemannn** [comment]
Should be fine then! **Approved**!

**Pull request merged**